console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx

```diff
@@ -139,10 +150,31 @@ export default function DeviceDetailPage() {
      */
   };
 
-  const onChangeDeviceStatus = () => {
-    /**
-     * TODO: implement change device status logic
-     */
+  const onConfirmToggleDeviceStatus = async (data: any) => {
+    const variable = {
+      a: { name: projectname, namespace },
+      b: [data.deviceId],
+      status: data.status
+    };
+    await setUpdateDeviceQueryVariables(variable);
+  };
+
+  const onChangeDeviceStatus = (status: boolean) => {
+    const dialogType: string = status
+      ? DialogTypes.ENABLE
+      : DialogTypes.DISABLE;
+    dispatch({
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.UPDATE_DEVICE_STATUS,
+      modalProps: {
+        onConfirm: onConfirmToggleDeviceStatus,
+        selectedItems: [deviceId],
+        option: DialogTypes.DISABLE,
+        data: { deviceId, status },
+        detail: getDetailForDialog([{ deviceId }], dialogType),
+        header: getHeaderForDialog([{ deviceId }], dialogType)
```

Comment:
To be consistent with other screens the confirm button should be in red color. Please add 
`iconType: "danger",
 confirmButtonLabel:"Enable" or "Disable" ` in the modalProps

Reply:
Do we really need a danger icon to change status of device?